pkg/controller/servicebindingrequest/secret.go

```diff
@@ -52,21 +54,61 @@ func (s *Secret) createOrUpdate(payload map[string][]byte) (*unstructured.Unstru
 	ns := s.plan.Ns
 	name := s.plan.Name
 	logger := s.logger.WithValues("Namespace", ns, "Name", name)
-	secretObj := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: ns,
-			Name:      name,
-		},
-		Data: payload,
+
+	var bindingRef interface{}
+	resource := SecretResource
+
+	if s.plan.SBR.Spec.BindingReference == nil {
+		return nil, err.New("BindingReference is empty")
+	}
+
+	gvk := corev1.SchemeGroupVersion.WithKind(s.plan.SBR.Spec.BindingReference.ObjectType.Kind)
+	if s.plan.SBR.Spec.BindingReference.ObjectType.Kind == ConfigMapKind {
+		resource = ConfigMapResource
 	}
 
-	gvk := corev1.SchemeGroupVersion.WithKind(SecretKind)
-	u, err := converter.ToUnstructuredAsGVK(secretObj, gvk)
+	objectMeta := metav1.ObjectMeta{
+		Namespace: ns,
+		Name:      name,
+	}
+
+	/*
```

Comment:
Do we need this commented portion?

Reply:
I was trying to announce what I was trying to do, and what the issue I was seeing with that :)